tests/Proto.Actor.Tests/SupervisionTests.cs

```diff
@@ -0,0 +1,121 @@
+                return Actor.Done;
+            }
+        }
+
+        class ChildActor : IActor
+        {
+
+            public Task ReceiveAsync(IContext context)
+            {
+                switch (context.Message)
+                {
+                    case string _:
+                        throw new Exception();
+                }
+                return Actor.Done;
+            }
+        }
+
+        [Fact]
+        public void OneForOneStrategy_Should_ResumeChildOnFailure()
+        {
+            var childMailboxStats = new TestMailboxStatistics(msg => msg is ResumeMailbox);
+            var strategy = new OneForOneStrategy((pid, reason) => SupervisorDirective.Resume, 1, null);
+            var childProps = Actor.FromProducer(() => new ChildActor())
+                .WithMailbox(() => UnboundedMailbox.Create(childMailboxStats));
+            var parentProps = Actor.FromProducer(() => new ParentActor(childProps))
+                .WithSupervisor(strategy);
+            var parent = Actor.Spawn(parentProps);
+
+            parent.Tell("hello");
+
+            childMailboxStats.Reset.Wait(1000);
+            Assert.Contains(ResumeMailbox.Instance, childMailboxStats.Posted);
+            Assert.Contains(ResumeMailbox.Instance, childMailboxStats.Received);
+        }
+
+        [Fact]
+        public void OneForOneStrategy_Should_StopChildOnFailure()
+        {
+            var childMailboxStats = new TestMailboxStatistics(msg => msg is Stopped);
+            var strategy = new OneForOneStrategy((pid, reason) => SupervisorDirective.Stop, 1, null);
+            var childProps = Actor.FromProducer(() => new ChildActor())
+                .WithMailbox(() => UnboundedMailbox.Create(childMailboxStats));
+            var parentProps = Actor.FromProducer(() => new ParentActor(childProps))
+                .WithSupervisor(strategy);
+            var parent = Actor.Spawn(parentProps);
+
+            parent.Tell("hello");
+
+            childMailboxStats.Reset.Wait(1000);
+            Assert.Contains(Stop.Instance, childMailboxStats.Posted);
+            Assert.Contains(Stop.Instance, childMailboxStats.Received);
+        }
+
+        [Fact]
+        public void OneForOneStrategy_Should_RestartChildOnFailure()
+        {
+            var childMailboxStats = new TestMailboxStatistics(msg => msg is Stopped);
+            var strategy = new OneForOneStrategy((pid, reason) => SupervisorDirective.Restart, 1, null);
+            var childProps = Actor.FromProducer(() => new ChildActor())
+                .WithMailbox(() => UnboundedMailbox.Create(childMailboxStats));
+            var parentProps = Actor.FromProducer(() => new ParentActor(childProps))
+                .WithSupervisor(strategy);
+            var parent = Actor.Spawn(parentProps);
+
+            parent.Tell("hello");
+
+            childMailboxStats.Reset.Wait(1000);
+            Assert.Contains(Restart.Instance, childMailboxStats.Posted);
```

Comment:
Yes, that would be very nice, especially for users if we can make it into a testing package.